Libraries/Opc.Ua.Client/Session/SessionConfiguration.cs

```diff
@@ -44,101 +52,142 @@ namespace Opc.Ua.Client
     [KnownType(typeof(X509IdentityToken))]
     [KnownType(typeof(IssuedIdentityToken))]
     [KnownType(typeof(UserIdentity))]
-    public class SessionConfiguration
+    public record class SessionOptions
     {
         /// <summary>
-        /// Creates a session configuration
+        /// The session name used by the client.
+        /// </summary>
+        [DataMember(IsRequired = true, Order = 20)]
+        public string? SessionName { get; set; }
```

Comment:
If required, we should not allow null, so the type should not be nullable. same for the other required properties.